lib/rules/no-warning-comments.js

```diff
@@ -135,18 +144,39 @@ module.exports = {
          * @returns {void} undefined.
          */
         function checkComment(node) {
-            if (astUtils.isDirectiveComment(node) && selfConfigRegEx.test(node.value)) {
+            if (
+                astUtils.isDirectiveComment(node) &&
+                selfConfigRegEx.test(node.value)
+            ) {
                 return;
             }
 
             const matches = commentContainsWarningTerm(node.value);
 
-            matches.forEach(matchedTerm => {
+            matches.forEach(({ type, comment }) => {
+                let commentToDisplay = "";
+                let truncated = false;
+
+                for (const c of comment.trim().split(/\s+/u)) {
+                    commentToDisplay = commentToDisplay.trim();
+                    const tmp = `${commentToDisplay} ${c}`;
+
+                    if (tmp.length <= CHAR_LIMIT) {
+                        commentToDisplay = tmp;
+                    } else {
+                        truncated = true;
+                        break;
+                    }
+                }
+
                 context.report({
                     node,
                     messageId: "unexpectedComment",
                     data: {
-                        matchedTerm
+                        type,
+                        comment: `${commentToDisplay.trim()}${
```

Comment:
If we change `const tmp = commentToDisplay ? ...` per the previous comment, I think that this `trim` will be also unnecessary.